### PR TITLE
Soften Nokogiri dependency to versions ">= 1.4"

### PR DESCRIPTION
**Pull request by JuanitoFatas**
Many Ruby / Rails projects already use Nokogiri 1.6.6.x series.
`<= 1.6.5` constraint make these projects impossible to integrate html-pipeline.
This pull request uses the workaround mentioned in https://github.com/sparklemotion/nokogiri/issues/1233#issuecomment-71462081.

If this change is okay, then will need:

- [x] an CHANGELOG entry + Release 2.2.1
- [x] ~~Backport this change to 1.11.0, Release 1.11.1 + an CHANGELOG entry~~ Found out no need to do this because at v1.10.0, the Nokogiri dependency is locked at [~> 1.4](https://github.com/jch/html-pipeline/blob/v1.11.0/html-pipeline.gemspec#L18).

--

Related: #176

### CODE REVIEW

File: lib/html/pipeline/@mention_filter.rb [hunk 1]
```diff
@@ -70,7 +70,7 @@ def self.mentioned_logins_in(text, username_pattern=UsernamePattern)
       def call
         result[:mentioned_usernames] ||= []
 
-        doc.search('text()').each do |node|
+        doc.search('.//text() | text()').each do |node|
```
Comment (jch) [inline lib/html/pipeline/@mention_filter.rb]: Why this change here? Since the tests pass, the behavior is the same, but I'm curious if this is the preferred convention.

**Comment by jch** [comment]
@JuanitoFatas :+1: to your task list. Sorry it took a while for me to get around to this.

File: lib/html/pipeline/@mention_filter.rb (hunk 1 above)
Comment (JuanitoFatas) [inline lib/html/pipeline/@mention_filter.rb]: Without this change, tests failed with Nokogiri 1.6.6.x:

```
Nokogiri::XML::XPath::SyntaxError:
Invalid expression: .//child::text() | self::child::text()
```

> Please, as a workaround, use #xpath with a real XPath expression (as above) to avoid having to convert CSS into a hacky XPath expression to work around unrelated DocumentFragment issues.
> https://github.com/sparklemotion/nokogiri/issues/1233#issuecomment-71465416

Nokogiri author suggested use real XPath to avoid this.

**Comment by alexandrnikitin** [comment]
@JuanitoFatas Thanks for the PR. This one is actual for me. I cannot use the latest `Jekyll` with `Jemoji` plugin because of that. Can we merge it please?

**Comment by JuanitoFatas** [comment]
The CI failure is due to failed to install Nokogiri 1.4.7.

@jch I wrote [a script](https://github.com/jch/html-pipeline/blob/feature/soften-nokogiri-dependency/bin/test-nokogiris) to show that this pull request won't failed in most Nokogiri versions, as per https://github.com/jch/html-pipeline/pull/208#discussion_r35469491.

Do you think it is worth to setup https://github.com/thoughtbot/appraisal for testing many nokogiri versions? 

If you think this Pull Request is good, I will remove 39d9488 and 940959a. Then finish my todo list listed in https://github.com/jch/html-pipeline/pull/208#issue-96257031.

**Comment by jch** [comment]
@JuanitoFatas :+1: to getting this PR merged. There's some test failures in travis you should fix before merging, but I think your plan is good.

> Do you think it is worth to setup https://github.com/thoughtbot/appraisal for testing many nokogiri versions?

You could open a separate PR to experiment with it. I'm open to it.

**Comment by JuanitoFatas** [comment]
@jch I think is good to merge and release. I'll work on the apprasial thing for testing multiple Nokogiris this weekend. Thank you and sorry this took so long...

**Comment by jch** [comment]
> Thank you and sorry this took so long...

No, thank **you**. It's open source, and I'm glad to you as a contributor :wink:
